src/java/org/apache/cassandra/db/streaming/CassandraStreamManager.java

```diff
@@ -104,7 +104,7 @@ public Predicate<SSTableReader> getSSTablePredicateForKeyspaceRanges(ReconciledK
 
             // if it's not repaired and there are no offsets, it was probably written before the table was using
             // mutation tracking and therefore should be considered unreconciled
-            if (sstableOffsets.isEmpty())
+            if (sstableOffsets.mutations().isEmpty() && sstableOffsets.transfers().isEmpty())
```

Comment:
It might be a nice to still have an overall `isEmpty()` method that covers both and leaves this class unchanged.